src/SourceMap.js

```diff
@@ -0,0 +1,163 @@
+// @flow
+import type {
+  ParsedMap,
+  VLQMap,
+  SourceMapStringifyOptions,
+  IndexedMapping,
+} from "./types";
+
+import path from "path";
+import { generateInlineMap, partialVlqMapToSourceMap } from "./utils";
+
+export default class SourceMap {
+  sourceMapInstance: any;
+
+  addEmptyMap(
+    sourceName: string,
+    sourceContent: string,
+    lineOffset: number = 0
+  ) {
+    this.sourceMapInstance.addEmptyMap(sourceName, sourceContent, lineOffset);
+    return this;
+  }
+
+  addRawMappings(
+    mappings: string,
+    sources: Array<string>,
+    names: Array<string>,
+    lineOffset: number = 0,
+    columnOffset: number = 0
+  ) {
+    this.sourceMapInstance.addRawMappings(
+      mappings,
+      sources,
+      names,
+      lineOffset,
+      columnOffset
+    );
+    return this;
+  }
+
+  addBufferMappings(
+    buffer: Buffer,
+    lineOffset: number = 0,
+    columnOffset: number = 0
+  ) {
+    this.sourceMapInstance.addBufferMappings(buffer, lineOffset, columnOffset);
+    return this;
+  }
+
+  addIndexedMapping(
+    mapping: IndexedMapping<string>,
+    lineOffset?: number = 0,
+    columnOffset?: number = 0
+  ) {
+    let hasValidOriginal =
+      mapping.original &&
+      typeof mapping.original.line === "number" &&
+      !isNaN(mapping.original.line) &&
+      typeof mapping.original.column === "number" &&
+      !isNaN(mapping.original.column);
+
+    this.sourceMapInstance.addIndexedMapping(
+      mapping.generated.line + lineOffset - 1,
+      mapping.generated.column + columnOffset,
+      // $FlowFixMe
+      hasValidOriginal ? mapping.original.line - 1 : -1,
+      // $FlowFixMe
+      hasValidOriginal ? mapping.original.column : -1,
+      mapping.source || "",
+      mapping.name || ""
+    );
+  }
+
+  // line numbers start at 1 so we have the same api as `source-map` by mozilla
+  addIndexedMappings(
+    mappings: Array<IndexedMapping<string>>,
+    lineOffset?: number = 0,
+    columnOffset?: number = 0
+  ) {
+    for (let mapping of mappings) {
+      this.addIndexedMapping(mapping, lineOffset, columnOffset);
+    }
+    return this;
+  }
+
+  addName(name: string): number {
+    return this.sourceMapInstance.addName(name);
+  }
+
+  addNames(names: Array<string>): Array<number> {
+    return names.map((n) => this.addName(n));
+  }
+
+  addSource(source: string): number {
+    return this.sourceMapInstance.addSource(source);
+  }
+
+  addSources(sources: Array<string>): Array<number> {
+    return sources.map((s) => this.addSource(s));
+  }
+
+  getSourceIndex(source: string): number {
+    return this.sourceMapInstance.getSourceIndex(source);
+  }
+
+  getSource(index: number): string {
+    return this.sourceMapInstance.getSource(index);
+  }
+
+  getNameIndex(name: string): number {
+    return this.sourceMapInstance.getNameIndex(name);
+  }
+
+  getName(index: number): string {
+    return this.sourceMapInstance.getName(index);
+  }
+
+  mappingWithIndexesToMappingWithStrings(
```

Comment:
Given it's called `addIndexedMappings`, I would have rather expected something like `indexedMappingToStringMapping `

Reply:
Sure, will change it